src/qt/evtloop.cpp

```diff
@@ -21,33 +21,48 @@
 
 #include <QtWidgets/QApplication>
 
-class wxQtIdleTimer : public QTimer
+class wxQtIdleTimer : public QTimer, public wxRefCounter
 {
 
 public:
-    wxQtIdleTimer( wxQtEventLoopBase *eventLoop );
+    wxQtIdleTimer();
+    ~wxQtIdleTimer();
     virtual bool eventFilter( QObject * watched, QEvent * event  );
 
 private:
     void idle();
-
-private:
-    wxQtEventLoopBase *m_eventLoop;
+    void ScheduleIdleCheck();
 };
 
-wxQtIdleTimer::wxQtIdleTimer( wxQtEventLoopBase *eventLoop )
+wxQtIdleTimer::wxQtIdleTimer()
 {
-    m_eventLoop = eventLoop;
+    // We need a QCoreApplication for event loops, create it here if it doesn't
+    // already exist as we can't modify wxAppConsole
+    if ( !QCoreApplication::instance() )
+    {
+        new QApplication(wxAppConsole::GetInstance()->argc, wxAppConsole::GetInstance()->argv);
```

Comment:
Hmm, when is this going to be destroyed?

Reply:
I moved this code from the event loop’s ctor.  I agree that it’s a bit wired. The act of creating the QApplication sets QCoreApplication::instance (hence the use of qApp further in the file) I believe this is cleaned up by wxApp.

I’m not happy with this but I wanted to leave this code undisdurbed until the event loop was “correct”

Reply:
I will have a look at this now that the original change request has been merged.  It wasn't obvious why this code was originally in the wxEventLoop c'tor but I wanted to limit the impact of my change.  

My thought would be to remove the app creation from event loop altogether and leave it entirely managed by wxApp